### PR TITLE
Adding PyEBSDIndex

### PR DESCRIPTION
**Pull request by hakonanes**
Signed-off-by: Håkon Wiik Ånes <hwaanes@gmail.com>

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [n/a] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by drowenhorst-nrl** [comment]
@hakonanes can confirm that I would like to be added as a maintainer.

**Comment by ocefpaf** [comment]
> @conda-forge/staged-recipes I didn't include a `about: license` entry since the license is a custom one (upstream license file [here](https://github.com/USNavalResearchLaboratory/PyEBSDIndex/blob/main/License)). Otherwise, this feedstock should be ready for review.

The SPDX for custom liceses is `LicenseRef-LICENSE-NAME`. You can add it that way to make the linter happy.

**Comment by hakonanes** [comment]
Thank you very much! I named it `LicenseRef-CUSTOM`.

File: recipes/pyebsdindex/meta.yaml [hunk 1]
```diff
@@ -48,6 +48,7 @@ test:
 about:
   home: https://pyebsdindex.readthedocs.io
   license_file: License
+  license: LicenseRef-CUSTOM
```
Comment (ocefpaf) [inline recipes/pyebsdindex/meta.yaml:51]: According to upstream it is Pubic Domain: https://github.com/USNavalResearchLaboratory/PyEBSDIndex/blob/main/License#L4

Also, if you choose a custom license it should be `LicenseRef-PyEBSDIndex`.

BTW, Public Domain is not in SPDX b/c it is, in a way, not a license, if you chose that it should be `LicenseRef-Public-Domain`.

**Comment by hakonanes** [comment]
Thank you for the clarification, I changed it to `LicenseRef-PyEBSDIndex`.

This feedstock depends on ten upstream packages (six BSD2/3, three MIT, one Apache 2), so we cannot use the `LicenseRef-Public-Domain` as far as I can tell.

File: recipes/pyebsdindex/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,59 @@
+{% set name = "pyebsdindex" %}
+{% set version = "0.1rc2" %}
```
Comment (ocefpaf) [inline recipes/pyebsdindex/meta.yaml:2]: I missed this one but a release candidate falls under: https://github.com/conda-forge/cfep/blob/main/cfep-05.md

Comment (hakonanes) [reply recipes/pyebsdindex/meta.yaml:2]: Did not know this, thanks for informing me of the prerelease specification for conda-forge.

PyEBSDIndex has only prereleases [available from PyPI](https://pypi.org/project/pyebsdindex/#history), and according to the [Exceptions part](https://pypi.org/project/pyebsdindex/#history) of the specification, packages without non-prereleases should *not* publish to a prerelease label. Correct?

Comment (ocefpaf) [reply recipes/pyebsdindex/meta.yaml:2]: To be fair that exception was made for black b/c it was a popular package that never had an official release up until recent. Some people find it funny to make it a forever beta/alpha product, I guess it reminds then on 1995s Google.

Anyway, your options now are:

- make a case that your package is as important as black to the community and that it must be in main channel
- add a conda_build_config.yaml and release it under a dev channel
- ask upstream for a stable release

Comment (hakonanes) [reply recipes/pyebsdindex/meta.yaml:2]: > that exception was made for black b/c it was a popular package that never had an official release up until recent

Thanks for the clarification. I understand that such an exception was made for black. I did *not* read the "Exceptions" part as to pertain only to *popular* packages, but every package with nothing else than prererelases. I might suggest the a more clear wording than "Certain packages" and "such a package".

I've adopted your second option.

Comment (ocefpaf) [reply recipes/pyebsdindex/meta.yaml:2]: Indeed the text leaves it open for more exceptions. That is the problem with exceptions, right!

Remember to remove that file and rerender when we have a stable release.

**Comment by hakonanes** [comment]
> Remember to remove that file and rerender when we have a stable release.

Will do. Thank you for your helpful and kind review, @ocefpaf!

**Pull request merged**